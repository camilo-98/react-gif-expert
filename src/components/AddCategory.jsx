import { useState} from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target})=>{

        setInputValue(target.value);
    }

    const onSubmit = (event)=>{
        event.preventDefault();
        const valueClean = inputValue.trim();
        if( valueClean.length <= 1 ) return;

        //setCategories(categories => [inputValue, ...categories]);
        onNewCategory(valueClean);
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            <input type="text" 
            placeholder="Buscar Gifs"
            value={ inputValue }
            onChange={ onInputChange }></input>
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequire
}