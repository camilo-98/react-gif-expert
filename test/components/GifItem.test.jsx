
import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem'; 

describe('Pruebas en <GiftItem/>', () => {

    const title = 'Black';
    const url = 'https://discogs.com/blacksabbath.jpg';

    test('Debe hacer match con el snapshot', () => {
    
        const { container } = render(<GifItem title={ title } url={ url } />);    
        expect(container).toMatchSnapshot();
    })
    
    test('Debe mostrar la imagen con el url y el alt indicado', () => {
    
        render(<GifItem title={ title } url={ url } />);
        //screen.debug();
        //expect( screen.getByRole('img').src ).toBe( url );
        //expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
 
    })

    test('Debe de mostrar el titulo en el componente', () => {
      
        render(<GifItem title={ title } url={ url } />);
        expect( screen.getByText(title) ).toBeTruthy();

    })
    

});
