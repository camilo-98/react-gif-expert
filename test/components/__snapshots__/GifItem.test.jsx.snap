// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Black"
      src="https://discogs.com/blacksabbath.jpg"
    />
    <p>
      Black
    </p>
  </div>
</div>
`;
